### DISCUSSION TITLE
How to handle failures in external scanner

### DISCUSSION THREAD

**Discussion by MichaHoffmann** [tree-sitter/tree-sitter#1191] [Q&A] [answered]
Hey everyone,

I was wondering how to handle failures in the external scanner ( for example: scanner size exceeds `TREE_SITTER_SERIALIZATION_BUFFER_SIZE` when serializing; allocation during a scan fails; some other precondition violation ). 
Since my C is not too strong i was wondering: is it bad form to use `assert` macros in the scanner code to handle such situations?

Thank you

**Comment by ObserverOfTime** [comment, marked as answer]
`assert()` or  `abort()`
